pkg/controller/shoot/shoot_control.go

```diff
@@ -227,42 +227,46 @@ func (c *defaultControl) ReconcileShoot(shootObj *gardenv1beta1.Shoot, key strin
 
 		// If we reach this line then the deletion timestamp is set, the confirmation annotation exists and its value matches the deletion
 		// timestamp itself. Consequently, it's safe to trigger the Shoot cluster deletion here.
-		c.recorder.Eventf(shoot, corev1.EventTypeNormal, gardenv1beta1.ShootEventDeleting, "[%s] Deleting Shoot cluster", operationID)
+		c.recorder.Eventf(shoot, corev1.EventTypeNormal, gardenv1beta1.EventDeleting, "[%s] Deleting Shoot cluster", operationID)
 		if updateErr := c.updateShootStatusDeleteStart(operation); updateErr != nil {
 			shootLogger.Errorf("Could not update the Shoot status after deletion start: %+v", updateErr)
 			return true, updateErr
 		}
 		if deleteErr := c.deleteShoot(operation); deleteErr != nil {
-			c.recorder.Eventf(shoot, corev1.EventTypeWarning, gardenv1beta1.ShootEventDeleteError, "[%s] %s", operationID, deleteErr.Description)
+			c.recorder.Eventf(shoot, corev1.EventTypeWarning, gardenv1beta1.EventDeleteError, "[%s] %s", operationID, deleteErr.Description)
 			if state, updateErr := c.updateShootStatusDeleteError(operation, deleteErr); updateErr != nil {
 				shootLogger.Errorf("Could not update the Shoot status after deletion error: %+v", updateErr)
 				return state != gardenv1beta1.ShootLastOperationStateFailed, updateErr
 			}
 			return true, errors.New(deleteErr.Description)
 		}
-		c.recorder.Eventf(shoot, corev1.EventTypeNormal, gardenv1beta1.ShootEventDeleted, "[%s] Deleted Shoot cluster", operationID)
+		c.recorder.Eventf(shoot, corev1.EventTypeNormal, gardenv1beta1.EventDeleted, "[%s] Deleted Shoot cluster", operationID)
 		if updateErr := c.updateShootStatusDeleteSuccess(operation); updateErr != nil {
 			shootLogger.Errorf("Could not update the Shoot status after deletion success: %+v", updateErr)
 			return true, updateErr
 		}
 		return false, nil
 	}
-
+	/* TODO This should be part of admission controller: Discuss with Rafael
```

Comment:
Good point, thank you. Please remove this TODO and file a separate PR. The check for non-set deletion timestamp of the seed should be done here: https://github.com/gardener/gardener/blob/master/plugin/pkg/shoot/seedmanager/admission.go#L117-L131